lib/isolated/Pattern/SingletonDecorator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace WooCommerce\PayPalCommerce\Isolated\Pattern;
```

Comment:
I tried solution 1 with `lib/packages/...` and `lib/isolated/...` but PSALM started giving errors even when switching back to `trunk`. Had to clear it's cache. 
`yarn ddev:psalm --clear-cache` `yarn ddev:psalm --clear-global-cache`.

Solution 2 of having `lib/Pattern` should work, but I don't know if the confusion it will generate will compensate.

What about having a `lib/common` `WooCommerce\PayPalCommerce\Common` for `Pattern/...` and other common stuff in the future? Alternative names:
* a) `lib/common` `WooCommerce\PayPalCommerce\Common`
* b) `lib/base` `WooCommerce\PayPalCommerce\Base`
* c) `lib/shared` `WooCommerce\PayPalCommerce\Shared`
* d) `lib/library` `WooCommerce\PayPalCommerce\Library`
* ...

Reply:
Yeah, `lib/common` seems ok.